pymodbus/client/base.py

```diff
@@ -121,7 +125,7 @@ async def async_execute(self, request) -> ModbusResponse:
                 f"ERROR: No response received after {self.retries} retries"
             )
 
-        return resp
+        return resp  # type: ignore[return-value]
```

Comment:
This is wrong, the problem is the typing of the method which should include "| None".

Reply:
Will fix